corehq/apps/linked_domain/ucr.py

```diff
@@ -34,6 +40,17 @@ def create_linked_ucr(domain_link, report_config_id):
     else:
         report_config = ReportConfiguration.get(report_config_id)
         datasource = DataSourceConfiguration.get(report_config.config_id)
+        if not report_config and not datasource:
```

Comment:
@snopoke I was trying to think of the best way to check if we are dealing with a  `RegistryReportConfiguration`, so here I'm assuming that if the `report_config` is a registry report config it won't be returned by `ReportConfiguration.get`. (Looking at this now the `and not datasource` is redundant. I'll remove that)

Reply:
I would recommend doing something like this (exception handling omitted):

```python
report_config = get_report_config_or_not_found(report_config_id)  # handle DocumentNotFound
datasource = report_config.config  # handle DataSourceConfigurationNotFoundError

if is_data_registry_report(report_config):
    try:
        DataRegistry.objects.accessible_to_domain(domain_link.linked_domain, slug=registry_slug)
    except DataRegistry.DoesNotExist:
        message = _("This report cannot be linked because the registry is not accessible to {}. Please make"
            " sure the registry invitation has been accepted before linking reports.").format(domain_link.linked_domain)
        raise RegistryNotAccessible(message)
```

Reply:
Nice, I didn't see that there was already a function for that.